src/compiler.ts

```diff
@@ -639,25 +649,84 @@ export class Compiler extends DiagnosticEmitter {
       }
     }
 
-    this.memoryOffset = memoryOffset;
+    // setup memory & table
+    this.initMemory(memoryOffset);
+    this.initTable();
```

Comment:
I think I would prefer to keep these as part of the function, since calling those functions happens exactly once and wouldn't do anything useful otherwise?

Reply:
I prefer split and group it into separate function. This allows you to firstly easily find the right code part, and secondly to move it up or down, which is sometimes very useful

Reply:
I also remember that the code here heavily relies on ordering, which was something I wanted to review, but can't reasonably do now :)

Reply:
Yes, I noticed that too, so I decided to group everything so that nothing got mixed up. In addition, the individual parts were scattered throughout the main initializing method, which also made it hard to understand what was going on